addons/fire/functions/fnc_burn.sqf

```diff
@@ -0,0 +1,245 @@
+            [1,1,1,-1],
+            [1,1,1,-1],
+            [1,1,1,-1],
+            [1,1,1,-1],
+            [1,1,1,0]
+        ],
+        [0.5,1],
+        1,
+        0,
+        "",
+        "",
+        _unit
+    ];
+
+    _fire setParticleRandom [
+        0.04 * _intensity,
+        ([0.1,0.1,0.1] vectorMultiply _intensity),
+        ([0.05,0.05,0.05] vectorMultiply _intensity),
+        0,
+        0.06 * _intensity,
+        [0,0,0,0],
+        0,
+        0
+    ];
+
+    private _color = 0.8 / _intensity;
+
+    _smoke setDropInterval (0.01 + 0.02 * _intensity);
+    _smoke setParticleParams [
+        ["\A3\data_f\ParticleEffects\Universal\Universal",16,7,48],
+        "",
+        "Billboard",
+        1,
+        3 * _intensity,
+        "destructionEffect1",
+        [0, 0, 0.5 * _intensity],
+        0,
+        0.05,
+        0.04,
+        0.05,
+        [0.5 * _intensity, 3 * _intensity],
+        [
+            [_color, _color, _color, 0.2],
+            [_color, _color, _color, 1],
+            [_color, _color, _color, 1],
+            [0.05 + _color, 0.05 + _color, 0.05 + _color, 0.9],
+            [0.1 + _color, 0.1 + _color, 0.1 + _color, 0.6],
+            [0.2 + _color, 0.2 + _color, 0.2 + _color, 0.3],
+            [1,1,1,0]
+        ],
+        [0.8,0.3,0.25],
+        1,
+        0,
+        "",
+        "",
+        _unit
+    ];
+
+    _smoke setParticleRandom [
+        0.7 * _intensity,
+        [1 - _intensity / 10, 1 - _intensity / 10, 1 - _intensity / 10],
+        [0.2 * _intensity, 0.2 * _intensity, 0.05 * _intensity],
+        0,
+        0.3,
+        [0.05,0.05,0.05,0],
+        0,
+        0
+    ];
+
+    _light setLightBrightness (_intensity / 30);
+
+    // pp effect
+    #ifdef ENABLE_PP_EFFECT
+    private _fnc_FlameEffect = {
+        params ["_unit", "_fnc_FlameEffect", "_counter"];
+
+        if (_unit == ACE_player) then {
+            [] call BIS_fnc_flamesEffect;
+        };
+
+        DEC(_counter);
+
+        if (_counter > 0) then {
+            [_fnc_FlameEffect, [_unit, _fnc_FlameEffect, _counter], 0.4] call CBA_fnc_waitAndExecute
+        };
+    };
+
+    [_unit, _fnc_FlameEffect, 8] call _fnc_FlameEffect; // recursive function
+    #endif
+
+    // propagate fire
+    {
+        if (local _x) then {
+            [_x, 2] call FUNC(burn);
+        };
+    } forEach (
+        nearestTerrainObjects [_unit, ["SMALL TREE", "BUSH"], PROPAGATE_RADIUS, false] +
+        nearestObjects [_unit, ["Man"], PROPAGATE_RADIUS]
+    );
+
+    // update for next cycle, only on local machine
+    if (local _unit) then {
+        private _timeStep = CBA_missionTime - _time;
+
+        // add damage
+        private _damage = damage _unit + FIRE_DPS * _timeStep;
+        private _damages = getAllHitPointsDamage _unit param [2, []] apply {_x + FIRE_DPS * _timeStep};
+
+        _unit setDamage _damage;
+
+        {
+            _unit setHitIndex [_forEachIndex, _x];
+        } forEach _damages;
```

Comment:
damage handling by passes the medical system. Should call the medical damage API using `fire` as the projectile type.